### DISCUSSION TITLE
Security

### DISCUSSION THREAD

**Discussion by PedroGuilhermeFariaDuarte** [nuxt/nuxt#22907] [Questions]
Hello team! 👋 

In the Nuxt 2 I see that $nuxt statement are available in window. I can access all store from application and call functions like mutation, commit and action...

In the nuxt 3, the same way is available but in appNuxt() function, my question is, Why?

All rule business, security layers, data, token and others are "storage" into store of application and can easily access in the Devtools from browser, one time I access a site from NFT and all pages that need token authorization was avaliable for me in  $nuxt statement.... 

**Sorry I post this in other category** 😅

**Comment by pi0** [comment]
Hi @PedroGuilhermeFariaDuarte. Regardless of whether exposing a direct function or not, client-side logic can be always accessed (and altered) by users. You should consider having your business and security layers on the server-side instead or consider using server component islands if it is impossible to separate the logic.
